tests/2-remote_branch_checker.bats

```diff
@@ -101,3 +101,16 @@ assert_prechecker () {
 @test "remote_branch_checker/remote_branch_checker.sh: grunt build failed" {
     assert_prechecker fixture-grunt-build-failed MDL-12345 cd4a6b8b0bca159d3abb1468794ed5a074c5b701
 }
+
+@test "remote_branch_checker/remote_branch_checker.sh: remote which doesnt exist" {
+    export branch="a-branch-which-will-never-exist"
```

Comment:
I did this because I am not testing the new code, I am testing the existing 'remote doesn't exist' code. Because thats what sort of surprised me about the way this stuff was not failing with a better error, when we had code to detect bad remotes. 

 I thought the test for the trimming the spaces would be a little bit misleading to adding it in.